pkg/collector/check/stats_test.go

```diff
@@ -0,0 +1,114 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package check
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	agentConfig "github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/telemetry"
+)
+
+// Mock Check implementation used for testing
+type mockCheck struct {
```

Comment:
to avoid having to redefine all the methods of the check interface, you'll want to embed the `StubCheck` struct from https://github.com/DataDog/datadog-agent/blob/605de63eadd419a0a644edd6c92b4f064efe000a/pkg/collector/check/stub.go#L15 in your `mockCheck` struct, and only define the methods you want to override.

Reply:
Great find (I didn't know about the stub) - fixed!